src/Integration/Notifications/NotificationIndicatorViewModel.cs

```diff
@@ -0,0 +1,175 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Windows.Input;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Integration.WPF;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.Integration.Notifications
+{
+    public class NotificationIndicatorViewModel : ViewModelBase, INotificationIndicatorViewModel
+    {
+        private string text;
+        private bool hasUnreadEvents;
+        private bool isVisible;
+        private bool isBalloonTooltipVisible;
+        private bool areNotificationsEnabled;
+        private readonly ITimer autocloseTimer;
+        private readonly Action<Action> uiThreadInvoker;
+
+        public ObservableCollection<SonarQubeNotification> NotificationEvents { get; }
+
+        // For testing
+        public NotificationIndicatorViewModel()
```

Comment:
Not sure to understand... If this ctor is only for testing why don't you move the `ThreadHelper.Generic.Invoke` action to the test itself?

Reply:
Misplaced comment, that should be on the next constructor. Sorry for confusion.